branches.go

```diff
@@ -50,7 +50,10 @@ func (b Branch) String() string {
 //
 // GitLab API docs:
 // https://docs.gitlab.com/ce/api/branches.html#list-repository-branches
-type ListBranchesOptions ListOptions
+type ListBranchesOptions struct {
+	ListOptions
+	Search string `url:"search,omitempty" json:"search,omitempty"`
```

Comment:
```suggestion
	Search *string `url:"search,omitempty" json:"search,omitempty"`
```

Reply:
I just tested the behaviour with this code:
```go
	opt := &gitlab.ListBranchesOptions{
		Search: "",
	}

	encodedJson, _ := json.Marshal(opt)
	values, _ := query.Values(opt)

	fmt.Println(string(encodedJson))
	fmt.Println(values.Encode())
```

Without the pointer it works as expected on empty string:
```text
{}

```

What kind of edge case do you see? I can't figure out circumstances that will lead to empty value appearance in query line or json.

Reply:
In this specific case you are right, but in general setting the field with an empty string is not the same as not setting the field at all. So to make the client be able to differentiate between those two, this need to be a pointer.